ctmds1/repository.py

```diff
@@ -0,0 +1,107 @@
+from typing import Optional, Dict
+import duckdb
+from .constants import Countries, Commodity
+
+
+async def init_db():
+    conn = duckdb.connect(database=":memory:")
+    conn.execute(
+        """
+        CREATE TABLE hourly_curve AS 
+        SELECT * FROM read_csv_auto('./ctmds1/data/hourly_curve_by_country_commodity.csv', HEADER=True)
+    """
+    )
+    conn.execute(
+        """
+        CREATE TABLE season_curve AS 
+        SELECT * FROM read_csv_auto('./ctmds1/data/curve_by_season.csv', HEADER=True)
+    """
+    )
+    conn.execute(
+        """
+        CREATE TABLE currency_rates AS 
+        SELECT * FROM read_csv_auto('./ctmds1/data/currency_rates.csv', HEADER=True)
+    """
+    )
+    return conn
+
+
+def get_hourly_curve_factor(
+    db, country: Countries, commodity: Commodity
+) -> Optional[Dict[int, float]]:
+    country_str = country.value
+    commodity_str = commodity.value
+
+    if (
```

Comment:
The FastAPI endpoint should already be doing this validation. Here you can assume the input is clean - shouldn't need these extra checks. Input validation is best performed closer to the point/s of user interaction.